### PR TITLE
Add the patient after completing the transfer

### PR DESCRIPTION
**Pull request by dmeijboom**
Also:

- Added more checks for possible state transitions
- Made sure notifications are handled even when the one before errored

### CODE REVIEW

File: domain/transfer/receiver/repository.go [hunk 1]
```diff
@@ -154,3 +155,31 @@ func (f repository) CreateOrUpdate(ctx context.Context, status, taskID string, c
...
+	}
+
+	const query = `SELECT * FROM incoming_transfers WHERE customer_id = :customer_id AND task_id = :task_id`
+
+	transfer := &sqlTransfer{
+		CustomerID: customerID,
+		TaskID:     fhirTaskID,
+	}
+
+	_, err = tx.NamedExecContext(ctx, query, transfer)
```
Comment (stevenvegt) [inline domain/transfer/receiver/repository.go:172]: I think you should use `tx.GetContext` for getting a value. This also gives back the `sql.ErrNoRows`.

File: domain/transfer/receiver/service.go [hunk 2]
```diff
@@ -41,12 +45,106 @@ func NewTransferService(authService auth.Service, localFHIRClientFactory fhir.Fa
...
+	}
+
+	// And at last, retrieve the patient and create it on the local FHIR server
+	patient := &resources.Patient{}
+
+	if err := client().ReadOne(ctx, fmt.Sprintf("/%s", fhir.FromStringPtr(composition.Subject.Reference)), patient); err != nil {
+		return fmt.Errorf("failed to read the patient: %w", err)
+	}
+
+	if err := s.localFHIRClientFactory().CreateOrUpdate(ctx, patient); err != nil {
```
Comment (stevenvegt) [inline domain/transfer/receiver/service.go:113]: I don't think you should use the same identifier as the remote patient.

**Review by stevenvegt — CHANGES_REQUESTED** [review]

File: domain/transfer/receiver/service.go (hunk 2 above)
Comment (dmeijboom) [reply domain/transfer/receiver/service.go:116]: I agree but what would be the alternative?

**Comment by dmeijboom** [comment]
Is blocked by: #128

**Pull request closed without merging**